File: moira/src/bridge/atropos.ts
import { invoke } from '@tauri-apps/api/core'

import type { RuntimeStatusPayload } from '@/bridge/contracts/atropos'
import type { WakeInputRequestPayload } from '@/bridge/contracts/clotho'

export async function fetchRuntimeStatus(): Promise<RuntimeStatusPayload> {
  return invoke<RuntimeStatusPayload>('runtime_status')
}

export async function wakeCore(request: WakeInputRequestPayload): Promise<RuntimeStatusPayload> {
  return invoke<RuntimeStatusPayload>('wake', { request })
}

export async function stopCore(): Promise<RuntimeStatusPayload> {
  return invoke<RuntimeStatusPayload>('stop')
}

export async function forceKillCore(): Promise<RuntimeStatusPayload> {
  return invoke<RuntimeStatusPayload>('force_kill')
}

File: moira/src-tauri/src/app/commands/atropos.rs
use tauri::State;

use moira_runtime::{atropos::model::RuntimeStatus, clotho::model::WakeInputRequest};

use crate::app::state::AppState;

#[tauri::command]
pub async fn runtime_status(state: State<'_, AppState>) -> Result<RuntimeStatus, String> {
    state.runtime.atropos().runtime_status().await
}

#[tauri::command]
pub async fn wake(
    request: WakeInputRequest,
    state: State<'_, AppState>,
) -> Result<RuntimeStatus, String> {
    state.runtime.atropos().wake(request).await
}

#[tauri::command]
pub async fn stop(state: State<'_, AppState>) -> Result<RuntimeStatus, String> {
    state.runtime.atropos().stop().await
}

#[tauri::command]
pub async fn force_kill(state: State<'_, AppState>) -> Result<RuntimeStatus, String> {
    state.runtime.atropos().force_kill().await
}
